app/src/app/classes/stix/stix-object.ts

```diff
@@ -410,6 +457,68 @@ export abstract class StixObject extends Serializable {
 
     }
 
+    /**
+     * parses given field for linked objects
+     * @param field the field in which to parse LinkByIds
+     */
+    private parseLinkByIds(field: string, restAPIService: RestApiConnectorService): Observable<LinkByIdParseResult> {
+        let reLinkById = /\(LinkById: (.*?)\)/gmu;
+        let links = field.match(reLinkById);
+        let result = new LinkByIdParseResult({
+            brokenLinks: this.validateBrokenLinks(field, [/\(LinkById:([^ ].*?)\)/gmu, /\(LinkByID:(.*?)\)/gmu, /\(linkById:(.*?)\)/gmu])
+        });
+
+        if (!links) return of(result); // no LinkByIds found
+
+        let link_map = {};
+        for (let link of links) {
+            let id = link.split("(LinkById: ")[1].slice(0, -1);
+            link_map[id] = this.findLink(id, restAPIService)
+        }
+
+        return forkJoin(link_map).pipe(
+            map((results) => {
+                let link_results = results as any;
+                for (let key of Object.keys(link_results)) {
+                    // verify whether or not all links were found
+                    if (!link_results[key]) result.missingLinks.add(key);
+                }
+                return result;
+            })
+        );
+    }
+
+    /**
+     * validate the given field for broken LinkByIds found via regex
+     * @param field field that may have LinkByIds
+     * @param {regex[]} regExes regular expressions matching potential invalid tags
+     */
+    private validateBrokenLinks(field: string, regExes): Set<string> {
+        let result = new Set<string>();
+        for (let regex of regExes) {
+            let brokenLinks = field.match(regex);
+            if (brokenLinks) {
+                brokenLinks.forEach(l => result.add(l));
+            }
+        }
+        return result;
+    }
+
+    /**
+     * retrieves the linked object by ID
+     * @param id the ID of the linked object
+     * @param restAPIService service to connect to the REST API
+     */
+    private findLink(id: string, restAPIService: RestApiConnectorService): Observable<boolean> {
+        return restAPIService.getAllObjects(null, null, null, null, true, true, true).pipe(
```

Comment:
findLink will call `getAllObjects` for every link by id.  We want to lazy load `getAllObjects` to avoid recalling this function that takes a few seconds. Another option is lazy loading by object type to not load objects that will not be used. See comment on line 25 of descriptive-view that covers the main idea of lazy loading by object type